### DISCUSSION TITLE
EnumValueObject

### DISCUSSION THREAD

**Discussion by xavierjohn** [vkhorikov/CSharpFunctionalExtensions#487] [Q&A]
Hi @vkhorikov, 
It is awesome to see the read-me updated.  Great overview when I show the library to others.

There is no mention about EnumValueObject and I recall there is something about it that you did not like. What would your ideal implementation of EnumValueObject look like? I want to incorporate your thinking.

Thanks.

**Comment by vkhorikov** [comment]
I wrote about them here: https://github.com/vkhorikov/CSharpFunctionalExtensions/issues/274#issuecomment-791078593
The idea is to simplify the implementation a bit and also change a couple of method names.
